File: csharp/DesignPattern.Interpreter/MulExpression.cs

namespace DesignPattern.Interpreter
{
    public class MulExpression: OperatorExpression
    {
        public MulExpression(Expression left, Expression right)
            : base(left, right)
        { 
        
        }

        public override double Interpret(Context context)
        {
            return this.left.Interpret(context) * this.right.Interpret(context);
        }
    }
}

File: c++/InterpreterPattern/MulExpression.cpp
#include "stdafx.h"
#include "MulExpression.h"


MulExpression::MulExpression(Expression *left,Expression *right) : OperatorExpression(left,right)
{

}


double MulExpression::Interpret(Context context)
{
	return this->left->Interpret(context) * this->right->Interpret(context);
}